### ISSUE TITLE
Matchmaker with Signalling Server (all with Https).

### ISSUE THREAD

**Issue by lcdlv** [EpicGames/PixelStreamingInfrastructure#225]
## Component your question relates to
- [ ] Signalling server
- [ ] Frontend library
- [ ] Frontend UI library
- [X] Matchmaker
- [ ] SFU
- [ ] This repo
- [ ] Other

## Background
Unreal 5.4.2 with Signalling Server 5.4 and Matchmaker 5.4.
I trying to configure a Matchmaker with 2 signalling server on the same machine, all signalling server need to be with Https.

## Question
I can access to my streaming with an only one Signalling server with http port 80 and https port 443. The problems comes with 2 at the same time with a matchmaker.

Matchmaker config.json:
```
{
	"HttpPort": 80,
	"UseHTTPS": false,
	"MatchmakerPort": 9999,
	"LogToFile": true,
	"EnableWebserver": true
}
```
Signalling server 1 config.json:
```
{
	"UseFrontend": false,
	"UseMatchmaker": true,
	"UseHTTPS": true,
	"LogToFile": true,
	"LogVerbose": true,
	"HomepageFile": "player.html",
	"AdditionalRoutes": {},
	"EnableWebserver": true,
	"MatchmakerAddress": "localhost",
	"MatchmakerPort": 9999,
	"PublicIp": "localhost",
	"HttpPort": 79,
	"HttpsPort": 442,
	"StreamerPort": 8888,
	"SFUPort": 7889,
	"MaxPlayerCount": -1
}
```
Signalling Server 2 config.json
```
{
	"UseFrontend": false,
	"UseMatchmaker": true,
	"UseHTTPS": true,
	"LogToFile": true,
	"LogVerbose": true,
	"HomepageFile": "player.html",
	"AdditionalRoutes": {},
	"EnableWebserver": true,
	"MatchmakerAddress": "localhost",
	"MatchmakerPort": 9999,
	"PublicIp": "localhost",
	"HttpPort": 81,
	"HttpsPort": 444,
	"StreamerPort": 8889,
	"SFUPort": 7000,
	"MaxPlayerCount": -1
}
```
When I open two games, those two signalling server and the matchmaker all seems to be fine:
```
Node directory found...skipping install.
Node version:
v18.17.0

up to date, audited 103 packages in 2s

14 packages are looking for funding
  run `npm fund` for details

2 high severity vulnerabilities

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
configFile config.json
Config: {
        "HttpPort": 80,
        "UseHTTPS": false,
        "MatchmakerPort": 9999,
        "LogToFile": true,
        "EnableWebserver": true
}
15:36:02.161 HTTP listening on *:80
15:36:02.162 Matchmaker listening on *:9999
15:36:38.636 Message TYPE: streamerConnected
15:36:38.637 Cirrus server localhost:442 ready for use
15:36:43.591 Message TYPE: streamerConnected
15:36:43.593 Cirrus server localhost:444 ready for use
15:36:45.200 Message TYPE: ping
```
The problem comes when I go to 127.0.0.1 or localhost, the matchmaker redirect to localhost:442 but the stream don't work.
If in one signalling server **change the https port to 443, this signalling server works, but I cannot put for all signalling server the same port.** 

What can I do to make the HTTPs work?

**Comment by mcottontensor** [comment]
I've created a PR here #227 that should address this issue. If you are able to pull those changes and test to see if it correctly fixes the issue that would be very helpful. Thanks.

**Comment by mcottontensor** [comment]
This has now been merged into 5.4, 5.3 and 5.2. Feel free to update your 5.4 branch and let me know if the issue is fixed.

**Comment by lcdlv** [comment]
Hi! The fix works perfectly!! Thanks!! 
Buuuut, here's another new problem. I need https, and now I get it, but my certificate it's for my domain and matchmaker are redirecting using the machine public IP, and the certificate (wich is for the domain) it's not working...
Example:
1) I go to my domain www.avatar.com
2) The matchmaker redirect me to the public IP
3) The signalling server work on https but the certificate not work.

Any idea to solve that?

Thanks!

**Comment by mcottontensor** [comment]
With the basic setup provided, each signalling server (more specifically the web server) will need it's own certificate since all the matchmaker does is forward the user to a known address. We have recently deprecated the matchmaker as things like this make it non-production ready and isn't really a recommended solution for scaling.

Because the signalling server and web server can technically be separated you might have better luck providing one web server that serves your frontend and have a modified matchmaker in the background to direct your websocket connections to various signalling servers. Or perhaps a reverse proxy setup with some kind of load balancing if you wanted to distribute the web serving workload.

Unfortunately we don't really have a production ready solution for this and so is a little outside the scope of what we provide at this time.

**Comment by lcdlv** [comment]
Thanks for the fix!! I will see what can I do. thanks! If you deprecate Matchmaker, how we will do what matchmaker attempt to do in the future?

**Comment by mcottontensor** [comment]
The matchmaker will continue to work as it currently does and will continue to be available from the 5.4 branches down. It will just no longer be supported or updated going forward.

**Comment by 1091515459** [comment]
To be honest, I have also encountered corresponding problems. Although you said that I need to find a solution by myself, I still hope that the official can provide corresponding solutions. With the popularity of cloud rendering, I think UE still needs to improve its ability in this aspect in the future
